src/utils/data-manipulation.ts

```diff
@@ -1,60 +1,262 @@
 import type { Prisma } from "@prisma/client";
-import type { UserIdAndAnswerId } from "~/models/types";
+import type {
+    DataByRoleAndQuestion,
+    AggregatedDataByRole,
+    Role,
+} from "~/models/types";
 import { db } from "~/server/db";
 
-function uniqueValues<T>(array: T[]): T[] {
-    return Array.from(new Set(array));
-}
-
-export const fetchUsersAndAnswerOptions = async (
-    userIds: string[],
-    answerIds: string[],
-) => {
-    return await Promise.all([
-        db.user.findMany({
-            where: { id: { in: userIds } },
+type QuestionResultWithRelations = Prisma.QuestionResultGetPayload<{
+    include: {
+        question: {
+            include: {
+                roles: true;
+            };
+        };
+        user: {
+            select: {
+                id: true;
+                name: true;
+                communicationPreferences: {
+                    select: { methods: true };
+                };
+                roles: {
+                    select: { role: true };
+                };
+            };
+        };
+        answer: {
             select: {
-                id: true,
-                name: true,
-                email: true,
-                communicationPreferences: true,
-                roles: true,
+                id: true;
+                option: true;
+            };
+        };
+    };
+}>;
+
+export const fetchAnswerData = async ({
```

Comment:
I think this method name can be improved, something like `retrieveAnswersByRole`